ngrams_across_time/experiments/run_circuit_discovery.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import plotly.graph_objects as go
+import argparse
+from tqdm import tqdm
+import pandas as pd
+import torch
+from einops import rearrange
+
+from auto_circuit.prune_algos.mask_gradient import mask_gradient_prune_scores
+from auto_circuit.data import PromptDataLoader
+from auto_circuit.types import AblationType
+from auto_circuit.visualize import net_viz
+from auto_circuit.utils.ablation_activations import batch_src_ablations
+from auto_circuit.utils.graph_utils import patch_mode
+
+
+from ngrams_across_time.experiments.load_models import load_models
+from ngrams_across_time.utils.utils import get_logit_diff
+
+
+
+def score_circuits(quantiles, edge_scores, source_model, rec_model, dataloader, label):
```

Comment:
nit: source and dest is more immediately understandable to me than source and rec